tasks/train-a/search.md

```diff
@@ -0,0 +1,374 @@
+
+- the number of rows of passenger seats
+- the number of seats to the left of the aisle in a row
+- the number of seats to the right of the aisle in a row
+
+A **route** is a set of connected stations that a train travels through with a specific list of carriages. A train can consist of carriages of different types in any order and in any quantity. A route does not have a direct association with travel time.
+
+A **schedule** or **ride** is a precisely defined time for a train’s journey between first and last stations on the route, indicating the _price for each type of carriage for each segment between stations_, so that passengers can pay only for the trip between the desired stations.
+
+Apart from ride, the **trip** is a journey between selected stations by user. It is part of ride.
+
+### Core elements
+
+- **Search section**:
+  - **Form field _From_**: Where the trip starts
+  - **Form field _To_**: Where the trip ends
+  - **Form field _Date_**: From what date to search
+    - **Search**: Button to initiate the search process
+- **Result list item**:
+  - **Arrival date/time**: When train reach the target
+  - **Departure date/time**: When train departs from the station
+  - **Duration**: Travel time
+  - **Carriage(s)**: Carriage type(s) with free seats and price
+
+### Requirements: Search section [(design 1)](#search-block-1)
+
+#### Requirement 1: Start City Field
+
+- The form must include a field to define the city where the trip starts.
+- This field must have an autocomplete function to assist users in selecting the city.
+
+#### Requirement 2: End City Field
+
+- The form must include a field to define the city where the trip ends.
+- This field must have an autocomplete function to assist users in selecting the city.
+
+#### Requirement 3: Date/Time Field
+
+- The form must include a calendar field to specify the minimal date and time of the trip.
+
+#### Requirement 4: Submit Button
+
+- The form must include a submit button.
+- The submit button must be disabled until the form is valid (all fields are correctly filled).
+
+### Acceptance Criteria: Search section [(design 1)](#search-block-1) (20)
+
+#### Acceptance Criteria 1: Start city field autocomplete (6)
+
+> [!WARNING]
+> **Technical quirk**  
+> Developer should obtain coordinates of the location throught whatever external services their want while user fill city name in the form field.
+
+- The form must include a field labeled "From" for the start city.
+- When the user begins typing in the "From" field, an autocomplete dropdown must appear with city suggestions.
+- The user must be able to select a city from the autocomplete suggestions.
+
+#### Acceptance Criteria 2: End city field autocomplete (6)
+
+> [!WARNING]
+> **Technical quirk**  
+> Developer should obtain coordinates of the location throught whatever external services their want while user fill city name in the form field.
```

Comment:
probably same typos here
`throught` probably should be `through`
and maybe `their want` > `they want`
and probably `user fills`